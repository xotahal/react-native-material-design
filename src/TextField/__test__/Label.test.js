import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from '../../styles/ThemeProvider.react';

import Label from '../Label.react';

/* eslint-env jest */

const text = 'test';
const props = {
    baseSize: 32,
    basePadding: 4,
    fontSize: 16,
    activeFontSize: 12,
    tintColor: 'blue',
    baseColor: 'black',
    errorColor: 'red',
    animationDuration: 225,
};

it('renders label', () => {
    const label = renderer
        .create(
            <ThemeProvider>
                <Label {...props}>{text}</Label>
            </ThemeProvider>);

    expect(label.toJSON())
        .toMatchSnapshot();
});

it('renders active label', () => {
    const label = renderer
        .create(
            <ThemeProvider>
                <Label active {...props}>{text}</Label>
            </ThemeProvider>);

    expect(label.toJSON())
        .toMatchSnapshot();
});

it('renders focused label', () => {
    const label = renderer
        .create(
            <ThemeProvider>
                <Label focused {...props}>{text}</Label>
            </ThemeProvider>);

    expect(label.toJSON())
        .toMatchSnapshot();
});

it('renders errored label', () => {
    const label = renderer
        .create(
            <ThemeProvider>
                <Label errored {...props}>{text}</Label>
            </ThemeProvider>);

    expect(label.toJSON())
        .toMatchSnapshot();
});

it('renders restricted label', () => {
    const label = renderer
        .create(
            <ThemeProvider>
                <Label restricted {...props}>{text}</Label>
            </ThemeProvider>);

    expect(label.toJSON())
        .toMatchSnapshot();
});
