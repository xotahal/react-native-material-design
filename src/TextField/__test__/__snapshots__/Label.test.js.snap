// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders active label 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "position": "absolute",
      "top": 16,
    }
  }
>
  <Text
    accessible={true}
    activeFontSize={12}
    allowFontScaling={true}
    baseColor="black"
    basePadding={4}
    baseSize={32}
    collapsable={undefined}
    ellipsizeMode="tail"
    errorColor="red"
    fontSize={16}
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.38)",
        "fontSize": 12,
      }
    }
    tintColor="blue"
  >
    test
  </Text>
</View>
`;

exports[`renders errored label 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "position": "absolute",
      "top": 36,
    }
  }
>
  <Text
    accessible={true}
    activeFontSize={12}
    allowFontScaling={true}
    baseColor="black"
    basePadding={4}
    baseSize={32}
    collapsable={undefined}
    ellipsizeMode="tail"
    errorColor="red"
    fontSize={16}
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(213, 0, 0, 1)",
        "fontSize": 16,
      }
    }
    tintColor="blue"
  >
    test
  </Text>
</View>
`;

exports[`renders focused label 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "position": "absolute",
      "top": 16,
    }
  }
>
  <Text
    accessible={true}
    activeFontSize={12}
    allowFontScaling={true}
    baseColor="black"
    basePadding={4}
    baseSize={32}
    collapsable={undefined}
    ellipsizeMode="tail"
    errorColor="red"
    fontSize={16}
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(33, 150, 243, 1)",
        "fontSize": 12,
      }
    }
    tintColor="blue"
  >
    test
  </Text>
</View>
`;

exports[`renders label 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "position": "absolute",
      "top": 36,
    }
  }
>
  <Text
    accessible={true}
    activeFontSize={12}
    allowFontScaling={true}
    baseColor="black"
    basePadding={4}
    baseSize={32}
    collapsable={undefined}
    ellipsizeMode="tail"
    errorColor="red"
    fontSize={16}
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.38)",
        "fontSize": 16,
      }
    }
    tintColor="blue"
  >
    test
  </Text>
</View>
`;

exports[`renders restricted label 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "position": "absolute",
      "top": 36,
    }
  }
>
  <Text
    accessible={true}
    activeFontSize={12}
    allowFontScaling={true}
    baseColor="black"
    basePadding={4}
    baseSize={32}
    collapsable={undefined}
    ellipsizeMode="tail"
    errorColor="red"
    fontSize={16}
    numberOfLines={1}
    style={
      Object {
        "color": "rgb(213, 0, 0)",
        "fontSize": 16,
      }
    }
    tintColor="blue"
  >
    test
  </Text>
</View>
`;
